### PR TITLE
[Feature-store] Fixes and enhancements in real-time flow

### PR DESCRIPTION
**Pull request by yaronha**
* fixes in real-time ingestion flow 
* fixes in model serving and real-time vector
* add TSDB Writer and writer options

### CODE REVIEW

File: mlrun/feature_store/api.py [hunk 1]
```diff
@@ -298,6 +298,9 @@ def infer_metadata(
     return df
 
 
+infer_metadata = infer
```
Comment (Hedingber) [inline mlrun/feature_store/api.py:301]: Add a comment that this is for BC

File: mlrun/datastore/sources.py [hunk 2]
```diff
@@ -206,12 +208,23 @@ def __init__(
...
-        super().__init__(name, path, attributes)
+        super().__init__(name, path, attributes, key_field, time_field)
         self.online = True
         self.workers = workers
 
+    def to_step(self, key_field=None, time_field=None):
+        import storey
+
+        return storey.Source(
+            key_field=self.key_field or key_field,
```
Comment (Hedingber) [inline mlrun/datastore/sources.py:223]: Shouldn't be the opposite ?
if we given a value use it, and if not use the `self.x` ?

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: mlrun/datastore/sources.py (hunk 2 above)
Comment (yaronha) [reply mlrun/datastore/sources.py:223]: no, the target class wins over the default which comes from the feature set spec

File: mlrun/feature_store/api.py (hunk 1 above)
Comment (yaronha) [reply mlrun/feature_store/api.py:301]: ok

**Review by Hedingber — APPROVED** [review]

**Pull request merged**